ProcessLib/ProcessVariable.h

```diff
@@ -87,14 +90,20 @@ class ProcessVariable
 		}
 	}
 
-	double getInitialConditionValue(MeshLib::Node const& n) const
+	double getInitialConditionValue(MeshLib::Node const& n,
+	                                int const component_id) const
 	{
-		return _initial_condition->getValue(n);
+		return _initial_condition->getValue(n, component_id);
 	}
 
+	// Get or create a property vector for results.
+	// The returned mesh property size is number of mesh nodes times tuple size.
+	MeshLib::PropertyVector<double>& getOrCreateMeshProperty();
+
 private:
 	std::string const _name;
-	MeshLib::Mesh const& _mesh;
+	MeshLib::Mesh& _mesh;
+	int _tuple_size;
```

Comment:
On cppcon15 it was recommended to use int. If there is need for large numbers, use long. And only if doing bit-arithmetic (or some other thing I forgot now, but probably smth. with addresses) use unsigned types.